parameter_sync.py

```diff
@@ -163,8 +186,54 @@ def get_parameters_with_prefix(prefix, next_token=None):
         logger.critical('AWS Secret Access Key not set - cannot continue')
 
     logger.debug('INIT')
-    logger.info('Getting parameters with prefix %s from AWS Parameter Store' % args.param_prefix)
-    logger.info('Parameter values will be compared against file contents in "%s" and updated if necessary' % args.cred_path)
+    logger.info(f'Getting parameters with prefix {args.param_prefix} from AWS Parameter Store')
+    logger.info(f'Parameter values will be compared against file contents in "{args.cred_path}" and updated if necessary')
     process_parameters_with_prefix(args.param_prefix, args.cred_path, args.aws_region,
                                    args.aws_access_key, args.aws_secret_key, args.dryrun)
+
+    if args.save_aws_creds:
+        if args.aws_access_key and args.aws_secret_key:
+            aws_access_key = args.aws_access_key
+            aws_secret_key = args.aws_secret_key
+        else:
+            aws_access_key = os.environ.get('AWS_ACCESS_KEY_ID')
+            aws_secret_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
+
+        if len(args.save_aws_creds) % 3 != 0:
```

Comment:
The idea here is that if you use the --save-aws-creds option, you need to provide three inputs, the filename to save to, the profile name and the region (or None). The first input is the filename, the second is the profile name, and the third is the region